plugins/reolink/src/reolink-api.ts

```diff
@@ -201,40 +213,59 @@ export class ReolinkCameraClient {
         return response.body?.[0]?.value?.DevInfo;
     }
 
-    private async ptzOp(op: string, speed: number) {
+    async getPtzPresets(): Promise<PtzPreset[]> {
         const url = new URL(`http://${this.host}/api.cgi`);
         const params = url.searchParams;
-        params.set('cmd', 'PtzCtrl');
+        params.set('cmd', 'GetPtzPreset');
+        const body = [
+            {
+                cmd: "GetPtzPreset",
+                action: 1,
+                param: {
+                    channel: this.channelId
+                }
+            }
+        ];
+        const response = await this.requestWithLogin({
+            url,
+            responseType: 'json',
+            method: 'POST'
+        }, this.createReadable(body));
+        return response.body?.[0]?.value?.PtzPreset?.filter(preset => preset.enable === 1);
+    }
 
-        const createReadable = (data: any) => {
-            const pt = new PassThrough();
-            pt.write(Buffer.from(JSON.stringify(data)));
-            pt.end();
-            return pt;
-        }
+    private async ptzOp(op: string, speed: number, id?: string) {
+        const url = new URL(`http://${this.host}/api.cgi`);
+        const params = url.searchParams;
+        params.set('cmd', 'PtzCtrl');
 
         const c1 = this.requestWithLogin({
             url,
             method: 'POST',
             responseType: 'text',
-        }, createReadable([
+        }, this.createReadable([
             {
                 cmd: "PtzCtrl",
                 param: {
                     channel: this.channelId,
                     op,
                     speed,
                     timeout: 1,
+                    id: id ? Number(id) : undefined
```

Comment:
make the id parameter itself a number and do the validation higher up.

Reply:
done